default.nix

```diff
@@ -11,14 +11,17 @@
 , config ? { allowUnfreePredicate = (import ./nix/lib/unfree.nix).unfreePredicate; }
   # Overrides for niv
 , sourcesOverride ? { }
-, packages ? import ./nix { inherit system crossSystem config sourcesOverride rev checkMaterialization enableHaskellProfiling; }
+, packages ? import ./nix { inherit system crossSystem config sourcesOverride rev checkMaterialization enableHaskellProfiling deferPluginErrors; }
   # An explicit git rev to use, passed when we are in Hydra
 , rev ? null
   # Whether to check that the pinned shas for haskell.nix are correct. We want this to be
   # false, generally, since it does more work, but we set it to true in the CI
 , checkMaterialization ? false
   # Whether to build our Haskell packages (and their dependencies) with profiling enabled.
 , enableHaskellProfiling ? false
+  # Whether to set the `defer-plugin-errors` flag on those packages that need
+  # it. If set to true, we will also build the haddocks for those packages.
+, deferPluginErrors ? false
```

Comment:
I don't think we need to bubble it up this far: this is only really for things that users (us) might want to set globally when building things. I think we probably just want to set this in the version we pass into the thing that makes the haddock.